pygmt/tests/test_x2sys_cross.py

```diff
@@ -0,0 +1,41 @@
+"""
+Tests for x2sys_cross
+"""
+import os
+from tempfile import TemporaryDirectory
+
+import pandas as pd
+import pytest
+
+from .. import which, x2sys_cross, x2sys_init
+
+# from ..datasets import load_sample_bathymetry
+# from ..exceptions import GMTInvalidInput
+
+
+@pytest.fixture(name="mock_x2sys_home")
+def fixture_mock_x2sys_home(monkeypatch):
+    """
+    Set the X2SYS_HOME environment variable to the current working directory
+    for the test session
+    """
+    monkeypatch.setenv("X2SYS_HOME", os.getcwd())
+
+
+def test_x2sys_cross_input_file(mock_x2sys_home):  # pylint: disable=unused-argument
+    """
+    Run x2sys_cross by passing in a filename
+    """
+    fname = which("@tut_ship.xyz", download="a")
+    with TemporaryDirectory(prefix="X2SYS", dir=os.getcwd()) as tmpdir:
+        tag = os.path.basename(tmpdir)
+        x2sys_init(tag=tag, fmtfile="xyz", force=True)
+        output = x2sys_cross(tracks=[fname], tag=tag, coe="i")
```

Comment:
Hmm, not quite sure why it's running into a segfault on all the CI services (it works fine on my computer). This is with just passing an actual filename, no virtualfile involved! The equivalent GMT command is:

```bash
export X2SYS_HOME=.
gmt x2sys_init GMT -Dxyz -F
gmt x2sys_cross @tut_ship.xyz -TGMT -Qi > tmp.txt
```

Reply:
The test crashes for me on macOS. I find it's annoying to have a segmental fault, but pytest doesn't show any debug messages.

I changed the test to a standalone python script and ran it locally. Here are the errors I got:
```
x2sys_cross [DEBUG]: gmt_get_filename: In: 0/360/-90/90 Out: 0/360/-90/90
x2sys_cross [DEBUG]: Look for file 0/360/-90/90 in /Users/seisman/.gmt
x2sys_cross [DEBUG]: Look for file 0/360/-90/90 in /Users/seisman/.gmt/cache
x2sys_cross [DEBUG]: Look for file 0/360/-90/90 in /Users/seisman/.gmt/server
x2sys_cross [DEBUG]: Got regular w/e/s/n for region (0/360/-90/90)
x2sys_cross [INFORMATION]: Files found: 1
x2sys_cross [INFORMATION]: Checking for duplicates :
x2sys_cross [INFORMATION]: 0 found
x2sys_cross [WARNING]: No time column, use dummy times
x2sys_cross [DEBUG]: Map distance calculation will be Cartesian
Not a valid unit: c [meter assumed]
Not a valid unit: c [meter assumed]
x2sys_cross [DEBUG]: gmtapi_init_export: Passed family = Data Table and geometry = Point
x2sys_cross [DEBUG]: Object ID 0 : Registered Data Table File /Users/seisman/Gits/gmt/pygmt/X2SYS8w1sgn5c/tmp_coe.txt as an Output resource with geometry Point [n_objects = 1]
x2sys_cross [DEBUG]: gmtapi_init_export: Added 1 new destination
x2sys_cross [DEBUG]: GMT_Init_IO: Returned first Output object ID = 0
x2sys_cross [DEBUG]: GMT_Begin_IO: Mode value 1 not considered (ignored)
x2sys_cross [DEBUG]: GMT_Begin_IO: Initialize record-by-record access for Output
x2sys_cross [DEBUG]: gmtapi_next_io_source: Selected object 0
x2sys_cross [INFORMATION]: Writing Data Table to file /Users/seisman/Gits/gmt/pygmt/X2SYS8w1sgn5c/tmp_coe.txt
x2sys_cross [DEBUG]: GMT_Begin_IO: Output resource access is now enabled [record-by-record]
x2sys_cross [DEBUG]: x2sys_get_data_path: Given track /Users/seisman/.gmt/cache/tut_s and suffix xyz
x2sys_cross [DEBUG]: x2sys_get_data_path: add_suffix gives T
x2sys_cross [DEBUG]: x2sys_get_data_path: Full path for /Users/seisman/.gmt/cache/tut_s will be /Users/seisman/.gmt/cache/tut_s.xyz
x2sys_cross [ERROR]: x2sys_read_file : Cannot open file /Users/seisman/.gmt/cache/tut_s.xyz
(null): Error from fclose [/Users/seisman/.gmt/cache/tut_ship]
```
Please check what makes no sense to you.

Reply:
The truncated path `~/.gmt/cache/tut_s.xyz` is strange, it should be `~/.gmt/cache/tut_ship.xyz`. Is it trying to cut off the last 3 characters? I'll try to investigate.

Reply:
Strangely the segfault goes away when using `@tut_ship.xyz` directly. Not sure why it doesn't like the output from `pygmt.which` :confused: Anyway, I'll continue with other stuff first.

Reply:
Ah it's interesting. It seems like the `x2sys_cross` command only works when we don't give the full paths (i.e. no slashes). So this would work:

```bash
gmt x2sys_cross @tut_ship.xyz -Qi -TGMT > tmp.xyz
```

but this won't

```bash 
gmt x2sys_cross ~/.gmt/cache/tut_ship.xyz -Qi -TGMT > tmp.xyz
```

This seems to be expected behaviour (though it will be annoying for Python users). I guess that's why we need to set an $X2SYS_HOME variable (see https://docs.generic-mapping-tools.org/6.1/supplements/x2sys/x2sys_cross.html).